### PR TITLE
Tooltip Title Improvements

### LINKED ISSUE #402: Question: Tooltip DateTime Formatting?
I could be wrong, but it seems like the `tooltip.dateFormat()` expects one of the `HOUR_DAY`, `DAY_MONTH`, etc formats. Is there a reason why we can't just pass an acceptable d3 formatting string instead? i.e.

`tooltip.dateFormat('%m/%d @ %H:%M %p')`

When I try to do the `MINUTE_HOUR` format, I get something like "Oct 18, 08 PM" — that's hardly a very friendly date format to use.

My ultimate question: **Is there any way currently to use a truly custom date format in the tooltips? Something more granular?**

Thanks!

### PR DESCRIPTION
**Pull request by rpheath**
This does 3 things for the tooltip titles:
* [BUG FIX]: adds ability to ONLY show the date in the title by setting the title to blank (before it still included the " - " prefix)
* [FEATURE]: adds the ability to use any custom date format for tooltip dates
* <strike>[CODE]: changes `shouldShowDateInTitle` accessor to `shouldAppendDateToTitle` to better reflect the behavior of what it actually does</strike>

## Examples
1. Date-Only Title
```javascript
tooltip.title('');
```
2. Custom Date Format
```javascript
tooltip
  .dateFormat(tooltip.axisTimeCombinations.CUSTOM)
  .dateCustomFormat('%m/%d @ %H:%M %p');
```

## Motivation and Context
I'm using Britecharts in a project, and these two things are pretty crucial to my ability to do so.

Before, when you set `tooltip.title('')` the title would look like this: ` - Aug 05, 03 AM`. This fix removes the prefixed " - " if the title is blank.

Additionally, I need to be able to show a more granular date in my tooltip titles. The existing options went as low as `MINUTE_HOUR`, but the format wasn't very user-friendly, to say the least: `Aug 05, 03 AM`. This allows me to use any custom format I desire.

## How Has This Been Tested?
I tested this by running Britecharts locally, including the docs to verify all API changes were present. Additionally, I've updated and added new specs where appropriate.

My testing environment is on macOS Sierra. The entire test suite passed with `SUCCESS`.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

This fixes #402.

### CODE REVIEW

**Comment by rpheath** [comment]
The tests pass, but it seems like there's a time zone issue I'm not accounting for when they're run on the CI server... I'm happy to update it if someone can steer me in the right direction on that.

File: src/charts/tooltip.js [hunk 1]
```diff
@@ -405,8 +406,12 @@ define(function(require){
         function updateTitle(dataPoint) {
             let tTitle = title;
```
Comment (Golodhros) [inline src/charts/tooltip.js]: Let's compute the date only one time:
  let formattedDate = formatDate(new Date(dataPoint[dateLabel]));
  ...

File: src/charts/tooltip.js [hunk 2]
```diff
@@ -595,6 +602,23 @@ define(function(require){
...
+         * Exposes the ability to use a custom date format
+         * @param  {String} _x          Desired custom format
+         * @return {String | module}  Current format or module to chain calls
+         * @public
+         * @example tooltip.dateFormat(tooltip.axisTimeCombinations.CUSTOM);
+         * tooltip.customDateFormat('%H:%M %p')
+         */
+        exports.customDateFormat = function(_x) {
+            if (!arguments.length) {
+              return customDateFormat || '%m/%d/%Y @ %H:%M %p';
```
Comment (Golodhros) [inline src/charts/tooltip.js]: I don't think we need this default here.
Nit: The indentation seems to be 2 spaces.

File: src/charts/tooltip.js [hunk 3]
```diff
@@ -677,22 +701,22 @@ define(function(require){
...
         /**
-         * Gets or Sets shouldShowDateInTitle
+         * Gets or Sets shouldAppendDateToTitle
          * @param  {Boolean} _x           Desired value
-         * @return {Boolean | module}    Current shouldShowDateInTitle or Chart module to chain calls
+         * @return {Boolean | module}    Current shouldAppendDateToTitle or Chart module to chain calls
          * @public
          */
-        exports.shouldShowDateInTitle = function(_x) {
+        exports.shouldAppendDateToTitle = function(_x) {
```
Comment (Golodhros) [inline src/charts/tooltip.js]: It seems to be duplicated!

File: test/specs/tooltip.spec.js [hunk 4]
```diff
@@ -140,6 +140,47 @@ define(['jquery', 'd3', 'tooltip'], function($, d3, tooltip) {
                         ).toBe('Tooltip title');
                     });
                 });
+
+                describe('when title is set to blank', function() {
+
+                    it('should only show the date of the tooltip', () =>  {
```
Comment (Golodhros) [inline test/specs/tooltip.spec.js]: Nice! Thanks!

File: src/charts/tooltip.js [hunk 5]
```diff
@@ -66,7 +66,7 @@ define(function(require){
             height = 45,
 
             title = 'Tooltip title',
-            shouldShowDateInTitle = true,
+            shouldAppendDateToTitle = true,
```
[the remark this answers is not in the document]
Comment (rpheath) [reply src/charts/tooltip.js]: Ahhh, yes, good call. For some reason I had it in my head this was part of the internal API. Reverted in 72d23876816a4cfcac43375b4214f342b11b0746.

File: src/charts/tooltip.js [hunk 6]
```diff
@@ -110,6 +110,7 @@ define(function(require){
 
             defaultAxisSettings = axisTimeCombinations.DAY_MONTH,
             dateFormat = null,
+            customDateFormat = null,
```
[the remark this answers is not in the document]
Comment (rpheath) [reply src/charts/tooltip.js]: Yes, good catch. Updated in 023de33843da28a09faa2be327f281302efffd61.

File: src/charts/tooltip.js (hunk 1 above)
Comment (rpheath) [reply src/charts/tooltip.js]: Duh... whoops! Updated in 24f951e4c0fe68860ff301452ffdd01c6960b41d.

File: src/charts/tooltip.js (hunk 2 above)
Comment (rpheath) [reply src/charts/tooltip.js]: Yeah, agreed. Fixed that and the spacing here: 4572fb93c437aff24095140383102233494cb06a.

File: src/charts/tooltip.js (hunk 3 above)
Comment (rpheath) [reply src/charts/tooltip.js]: NO IDEA how/why that happened, but removed it in 206e08e37f9cd7f47a7c00540a34d1d62e551bd3.

**Comment by rpheath** [comment]
No problem, happy to help! @Golodhros will there be an upcoming release with this PR merged in soon?

Thanks!

**Comment by Golodhros** [comment]
Yeah, pretty soon!

**Comment by rpheath** [comment]
Any update on the next release? Thanks!

**Pull request merged**